contracts/libraries/FullMath.sol

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.4.0;
+
+/// @title Contains 512-bit math functions
+/// @notice Facilitates multiplication and division that can have overflow of an intermediate value without any loss of precision
+/// @dev Handles "phantom overflow" i.e., allows multiplication and division where an intermediate value overflows 256 bits
+library FullMath {
```

Comment:
I think you should be able to import this from v3-core

Reply:
oh actually I see you were doing this first. Was something going wrong importing them?

Reply:
Removed